kmod/src/xattr.c

```diff
@@ -498,6 +518,52 @@ static int parse_totl_u64(const char *s, int len, u64 *res)
 	return kstrtoull(str, 0, res) != 0 ? -EINVAL : 0;
 }
 
+static int parse_totl_u32(const char *s, int len, unsigned long int *res)
+{
+	char str[SCOUTFS_XATTR_MAX_TOTL_U32 + 1];
```

Comment:
Instead of duplicating the totl_u64 implementaion with a smaller type, it can just call to parse it as a _u64 then return EINVAL if the parsed value is > U32_MAX.